components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1139,34 +1143,34 @@ _validate_deps() {
     # of the duplicated `ORIGIN/NAME` tokens. This will be used to star the
     # problematic dependencies in the graph.
     _dupes_qualified=$(echo "$tdeps" \
-      | egrep "($(echo "$dupes" | tr '\n' '|' | sed 's,|$,,'))")
+      | grep -E "($(echo "$dupes" | tr '\n' '|' | sed 's,|$,,'))")
 
     warn
     warn "The following runtime dependencies have more than one version"
     warn "release in the full dependency chain:"
     warn
-    echo "$dupes" | while read dupe; do
```

Comment:
Feels like this is a behavior change?

Reply:
It shouldn't be. It's just adding the `-r` flag to `read`, which I'm pretty sure we want in this case. See https://github.com/koalaman/shellcheck/wiki/SC2162#read-without--r-will-mangle-backslashes

Reply:
Gotcha.  I had no idea of the background behind `read` and the `-r` flag.  FWIW, I really appreciate all the context you've added throughout this PR about _why_ changes were made, with appropriate links to the shellcheck documentation (which is refreshingly lucid and comprehensive).